File: run/RadiationTransportFLD/FSRadPoint/plot_results.m
function [] = plot_results()
%
% usage:  plot_results()
%

% loop over time steps
tend = 10;
for it = 0:tend

   % set subdirectory to load files from
   if (it < 10) 
      dir = [ 'DD000' , num2str(it) ];
      stub = [ 'data000' , num2str(it) ];
   elseif (it < 100) 
      dir = [ 'DD00' , num2str(it) ];
      stub = [ 'data00' , num2str(it) ];
   elseif (it < 1000)
      dir = [ 'DD0' , num2str(it) ];
      stub = [ 'data0' , num2str(it) ];
   elseif (it < 10000)
      dir = [ 'DD' , num2str(it) ];
      stub = [ 'data' , num2str(it) ];
   else
      error([ 'Time step ',it,' is out of range']);
   end

   % get Units from parameter file
   pfile = [ dir, '/', stub ];
   fin = fopen(pfile,'r');
   if (fin < 0)
      error([ 'Could not open ',pfile,' for input']);
   end

   % examine parameter file for relevant fields
   %    left domain boundary
   for i=1:1000
      buffer = fgetl(fin);
      if (findstr(buffer,'DomainLeftEdge'))
	 [text,buffer] = strtok(buffer,'=');
	 [text,tmp] = strtok(buffer);
	 leftedge = str2num(tmp); frewind(fin); break
      end
   end
   %    right domain boundary
   for i=1:1000
      buffer = fgetl(fin);
      if (findstr(buffer,'DomainRightEdge'))
	 [text,buffer] = strtok(buffer,'=');
	 [text,tmp] = strtok(buffer);
	 rightedge = str2num(tmp); frewind(fin); break
      end
   end
   %    time
   for i=1:1000
      buffer = fgetl(fin);
      if (findstr(buffer,'InitialTime'))
	 [text,buffer] = strtok(buffer,'=');
	 [text,tmp] = strtok(buffer);
	 t = str2num(tmp); frewind(fin); break
      end
   end
   %    density units
   for i=1:1000
      buffer = fgetl(fin);
      if (findstr(buffer,'DensityUnits'))
	 [text,buffer] = strtok(buffer,'=');
	 [text,tmp] = strtok(buffer);
	 dUnits = str2num(tmp); frewind(fin); break
      end
   end
   %    length units
   for i=1:1000
      buffer = fgetl(fin);
      if (findstr(buffer,'LengthUnits'))
	 [text,buffer] = strtok(buffer,'=');
	 [text,tmp] = strtok(buffer);
	 lUnits = str2num(tmp); frewind(fin); break
      end
   end
   %    time units
   for i=1:1000
      buffer = fgetl(fin);
      if (findstr(buffer,'TimeUnits'))
	 [text,buffer] = strtok(buffer,'=');
	 [text,tmp] = strtok(buffer);
	 tUnits = str2num(tmp); frewind(fin); break
      end
   end
   fclose(fin);

   % set radiation units, box bounds, time
   rUnits = dUnits*(lUnits/tUnits)^2;
   leftedge = leftedge*lUnits;
   rightedge = rightedge*lUnits;
   t = t*tUnits;

   % set hdf5 name
   hfile = [ dir, '/', stub, '.cpu0000' ];
   gridname = [ 'Grid00000001' ];
   
   % read radiation field and re-scale
   name = [ gridname, '/FS_Radiation' ];
   tmp = double(hdf5read(hfile,name));
   E = tmp*rUnits;
   clear tmp
   
   % plot 2D slice at zlevel = 0
   [Nx,Ny,Nz] = size(E);
   xspan = linspace(leftedge(1),rightedge(1),Nx)'./(rightedge(1)-leftedge(1));
   yspan = linspace(leftedge(2),rightedge(2),Ny)'./(rightedge(2)-leftedge(2));
   figure(1), surf(yspan,xspan,log10(E(:,:,1)));
   shading interp, colorbar, axis equal, view(0,90)
%   figure(1), contour(yspan,xspan,log10(E(:,:,1)));
%   colorbar, axis equal, view(0,90)
   xlabel('y'), ylabel('x')
   title(sprintf('log E_{fs}, t = %g Tf',it/10),'FontSize',14);

   % plot 1D profile
   figure(2), plot(xspan,log10(E(:,1,1)));
   axis([0 1 -30 -20]);
   xlabel('x'), title(sprintf('log E_{fs}, t = %g Tf',it/10),'FontSize',14);

   % plot 1D profile
   figure(3), plot(xspan,E(:,1,1));
   axis([0 1 0 1e-20]);
   xlabel('x'), title(sprintf('E_{fs}, t = %g Tf',it/10),'FontSize',14);

   % plot scaled 1D profile
   figure(4), plot(xspan,E(:,1,1).*xspan.*xspan);
%   axis([0 1 0 5e-22]);
   xlabel('x'), title(sprintf('E_{fs}*r^2, t = %g Tf',it/10),'FontSize',14);

   % clean up
   clear E;

   % save plots from mid-way through run
   if (it == 5) 
      h = figure(1);
      saveas(h,'Econtour.fig');
      saveas(h,'Econtour.eps','epsc');
      saveas(h,'Econtour.png');
   
      h = figure(2);
      saveas(h,'Eprofile.fig');
      saveas(h,'Eprofile.eps','epsc');
      saveas(h,'Eprofile.png');

      h = figure(3);
      saveas(h,'Eprofile2.fig');
      saveas(h,'Eprofile2.eps','epsc');
      saveas(h,'Eprofile2.png');
end
   
   % hold on a second
   pause(1)
   
end  % time step loop

% end of file

File: run/RadiationTransportFLD/FSRadPoint/plot_results.py
# matplotlib-based plotting script for point-source streaming radiation test
# Daniel R. Reynolds, reynolds@smu.edu

import h5py
from pylab import *
import numpy as np

# set the graphics output type
pictype = '.png'
   
# define some helpful functions
def get_params(file):
    """Returns dUnit, tUnit, lUnit from a given parameter file"""
    import shlex
    f = open(file)
    for line in f:
        text = shlex.split(line)
        if ("DensityUnits" in text):
            dUnit = float(text[len(text)-1])
        elif ("TimeUnits" in text):
            tUnit = float(text[len(text)-1])
        elif ("LengthUnits" in text):
            lUnit = float(text[len(text)-1])
    return [dUnit, tUnit, lUnit]

def load_snapshot(tdump):
    """Returns x, E for a given data output"""
    import h5py
    import numpy as np
    sdump = repr(tdump).zfill(4)
    pfile = 'DD' + sdump + '/data' + sdump
    hfile = pfile + '.cpu0000'
    f = h5py.File(hfile,'r')
    Ef3D = f.get('/Grid00000001/FS_Radiation')
    dUnit, tUnit, lUnit = get_params(pfile)
    rUnit = dUnit*lUnit*lUnit/tUnit/tUnit
    nx, ny, nz = Ef3D.shape
    E = Ef3D[:][0][0]
    N = E.size
    x = linspace(0.01, 1.0, N)
    return [x, E]

def load_contour(tdump):
    """Returns E for a given data output"""
    import h5py
    import numpy as np
    sdump = repr(tdump).zfill(4)
    pfile = 'DD' + sdump + '/data' + sdump
    hfile = pfile + '.cpu0000'
    f = h5py.File(hfile,'r')
    Ef3D = f.get('/Grid00000001/FS_Radiation')
    dUnit, tUnit, lUnit = get_params(pfile)
    rUnit = dUnit*lUnit*lUnit/tUnit/tUnit
    nx, ny, nz = Ef3D.shape
    E = Ef3D[:][:][0]
    return [E]
    

# plot first snapshot
figure(1)
[x,E] = load_snapshot(0)
plot(x,E,'k-')
xlabel('x')
ylabel('E')
title('Streaming radiation history')
figure(2)
semilogy(x,E,'k-')
xlabel('x')
ylabel('E')
title('Streaming radiation history (log)')
figure(3)
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
plot(x,Escale,'k-')
xlabel('x')
ylabel('$4 \pi r^2 E$')
title('Scaled streaming radiation history')
figure(4)
semilogy(x,Escale,'k-')
xlabel('x')
ylabel('$4 \pi r^2 E$')
title('Scaled streaming radiation history (log)')

# repeat for next dataset
[x,E] = load_snapshot(1)
figure(1)
plot(x,E,'b-')
figure(2)
semilogy(x,E,'b-')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'b-')
figure(4)
semilogy(x,Escale,'b-')

# repeat for next dataset
[x,E] = load_snapshot(2)
figure(1)
plot(x,E,'g-')
figure(2)
semilogy(x,E,'g-')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'g-')
figure(4)
semilogy(x,Escale,'g-')

# repeat for next dataset
[x,E] = load_snapshot(3)
figure(1)
plot(x,E,'r-')
figure(2)
semilogy(x,E,'r-')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'r-')
figure(4)
semilogy(x,Escale,'r-')

# repeat for next dataset
[x,E] = load_snapshot(4)
figure(1)
plot(x,E,'c-')
figure(2)
semilogy(x,E,'c-')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'c-')
figure(4)
semilogy(x,Escale,'c-')

# repeat for next dataset
[x,E] = load_snapshot(5)
figure(1)
plot(x,E,'m-')
figure(2)
semilogy(x,E,'m-')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'m-')
figure(4)
semilogy(x,Escale,'m-')

# repeat for next dataset
[x,E] = load_snapshot(6)
figure(1)
plot(x,E,'y-')
figure(2)
semilogy(x,E,'y-')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'y-')
figure(4)
semilogy(x,Escale,'y-')

# repeat for next dataset
[x,E] = load_snapshot(7)
figure(1)
plot(x,E,'k-')
figure(2)
semilogy(x,E,'k-')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'k-')
figure(4)
semilogy(x,Escale,'k-')

# repeat for next dataset
[x,E] = load_snapshot(8)
figure(1)
plot(x,E,'b--')
figure(2)
semilogy(x,E,'b--')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'b--')
figure(4)
semilogy(x,Escale,'b--')

# repeat for next dataset
[x,E] = load_snapshot(9)
figure(1)
plot(x,E,'g--')
figure(2)
semilogy(x,E,'g--')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'g--')
figure(4)
semilogy(x,Escale,'g--')

# repeat for next dataset
[x,E] = load_snapshot(10)
figure(1)
plot(x,E,'r--')
figure(2)
semilogy(x,E,'r--')
Escale = np.multiply(np.multiply(E,x),x)*4.0*pi
figure(3)
plot(x,Escale,'r--')
figure(4)
semilogy(x,Escale,'r--')

# finish off plot and save to file
figure(1)
#axis([0.0, 1.0, -30.0, -14.0])
savefig('rad_snapshots' + pictype)
figure(2)
#axis([0.0, 1.0, 0.0, -15.0])
savefig('rad_snapshots_log' + pictype)
figure(3)
savefig('scaled_rad_snapshots' + pictype)
figure(4)
savefig('scaled_rad_snapshots_log' + pictype)


# plot contour at halfway point
[E] = load_contour(5)
nx, ny = E.shape
x = linspace(0.0,1.0,nx)
y = linspace(0.0,1.0,ny)
X, Y = meshgrid(x,y)
figure(5)
h = imshow(E, hold=False, extent=(0.0, 1.0, 0.0, 1.0), origin='lower')
colorbar(h)
title('E contour at T/2')
savefig('contour' + pictype)

figure(6)
h = imshow(log10(E), hold=False, extent=(0.0, 1.0, 0.0, 1.0), origin='lower')
colorbar(h)
title('log(E) contour at T/2')
savefig('contour_log' + pictype)


